File: src/nav_pkg/src/wp_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void NavResultCallback(const std_msgs::String &msg)
{
    ROS_WARN("[NavResultCallback] %s", msg.data.c_str());
}

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "wp_node");
    ros::NodeHandle nh;
    ros::Publisher nav_pub = nh.advertise<std_msgs::String>("/waterplus/navi_waypoint", 10);
    ros::Subscriber nav_sub = nh.subscribe("/waterplus/navi_result", 10, NavResultCallback);

    sleep(1);

    std_msgs::String nav_msg;
    nav_msg.data = "2";
    nav_pub.publish(nav_msg);

    ros::spin();

    return 0;
}

File: src/nav_pkg/scripts/wp_node.py
#!/usr/bin/env python3
# coding=utf-8

import rospy
from std_msgs.msg import String


def NavResultCallback(msg):
    rospy.loginfo("导航结果 = %s",msg.data)

if __name__ == "__main__":

    rospy.init_node("py_wp_node")
    navi_pub = rospy.Publisher("/waterplus/navi_waypoint", String, queue_size=10)

    result_sub = rospy.Subscriber("/waterplus/navi_result", String, NavResultCallback,queue_size=10)

    rospy.sleep(1)

    msg = String()
    msg.data = '3'

    navi_pub.publish(msg)

    rospy.spin()
